src/partition/repart.c

```diff
@@ -2066,7 +2099,7 @@ static int context_dump_partitions(Context *context, const char *node) {
                                 t,
                                 TABLE_STRING, gpt_partition_type_uuid_to_string_harder(p->type_uuid, uuid_buffer),
                                 TABLE_STRING, empty_to_null(label) ?: "-", TABLE_SET_COLOR, empty_to_null(label) ? NULL : ansi_grey(),
-                                TABLE_UUID, sd_id128_is_null(p->new_uuid) ? p->current_uuid : p->new_uuid,
+                                TABLE_UUID, !sd_id128_is_null(p->new_uuid) || p->new_uuid_is_set ? p->new_uuid : p->current_uuid,
```

Comment:
Why not simply use `p->new_uuid_is_set`? I think `!sd_id128_is_null()` is completely redundant.